p4-16/psa/PSA.mdk

```diff
@@ -146,49 +143,105 @@ of native and non-native operations. Target documentation should highlight such
 implementations, so that application programmers are aware of the
 potentially significant performance penalties.
 
-Semantics of behavior for multiple calls to PRE APIs
+The pseudocode below defines where copies of packets will be made
+after the Ingress control block has completed executing, based upon
+the contents of several metadata fields in the struct
+`psa_ingress_output_metadata_t`.
+
+```
+[INCLUDE=psa.p4:Metadata_ingress_output]
+
+    psa_ingress_output_metadata_t ostd;
 
-The semantics of calling the PRE APIs is equivalent to setting
-intrinsic metadata fields/bits and assuming that the PRE looks up the
-fields in the following order: drop, truncate, multicast, clone,
-output_port.
+    if (ostd.clone) {
+        create a copy of the packet to the clone target
```

Comment:
Good questions, and not the only ones I have about how all of this works, as is hopefully obvious from the TBD comments I have in the text.  I'm happy to add more TBD questions to this pull request, unless maybe @hanw has suggested answers he wants me to incorporate in this PR.

My hope in this PR was to at least first get a start at what the possibilities are for controlling what happens to a packet after it is finished with ingress or egress processing, the metadata fields that control it, and the pseudocode that defines at least the relative priorities of these actions, e.g. clone vs. drop vs. resubmit.